classes/file/FileManager.inc.php

```diff
@@ -543,21 +543,97 @@ function parseFileExtension($fileName) {
 
 	/**
 	 * Decompress passed gziped file.
-	 * @param $filePath string
-	 * @param $errorMsg string
-	 * @return boolean|string
+	 * @param $filePath string The compressed file's filename
+	 * @param $errorMsg string Any fatal error encountered (modified by reference)
+	 * @return boolean|string False on failure, or the decompressed filename on success.
 	 */
 	function decompressFile($filePath, &$errorMsg) {
+		if (extension_loaded('Zlib')) {
+			$chunk = 1024*1024;
+			$newFilePath = preg_replace('/[.]tgz$/i', '.tar.gz', $filePath);
+			$newFilePath = preg_replace('/[.]gz$/i', '', $newFilePath);
+			if ($newFilePath !== $filePath) {
+				$gz = gzopen($filePath, 'rb');
+				if ($gz) {
+					$file = fopen($newFilePath, 'wb');
+					if ($file) {
+						while (!gzeof($gz)) {
+							$data = gzread($gz, $chunk);
+							fwrite($file, $data);
+						}
+						fclose($file);
+						gzclose($gz);
+						unlink($filePath);
+						return $newFilePath;
+					}
+					fclose($file);
+				}
+			}
+		}
 		return $this->_executeGzip($filePath, true, $errorMsg);
 	}
 
+	
+	/**
+	 * Extract files from tgz or zip
+	 * @param $filePath string The compressed archive filename
+	 * @param $destination string The location to which to extract the contents
+	 * @param $errorMsg string Any fatal error encountered (modified by reference)
+	 * @return boolean Success
+	 */
+	function extractFiles($filePath, $destination, &$errorMsg) {
+		$success = false;
+		$errors = array();
+		if (extension_loaded('Phar')) {
```

Comment:
Do we have any `Phar` use cases?

Reply:
This is the only one I actually care about in all of this mess.  I'm not actually using `Phar` except for `PharData` which has convenient functions to decompress either Zip or TGZ archives in one handler.

Reply:
Ah, good, that makes sense. What do you think about a 3rd-party polyfill e.g. https://packagist.org/packages/alchemy/zippy (at a glance)?

Reply:
[appreciates the muttering sympathetically]

Reply:
Check the Packagist listing for each -- I often use the popularity numbers as a proxy for "how much outcry will there be if this thing goes away without an easy path forward".